### ISSUE TITLE
ServletHttpHandler static resources served by jetty order sensitive

### ISSUE THREAD

**Issue by mattmoss** [micronaut-projects/micronaut-servlet#240]
### Expected Behavior

The following config is problematic with jetty:
```yaml
micronaut:
  application:
    name: test-service
  router:
    static-resources:
      swagger:
        paths: classpath:META-INF/swagger
        mapping: /swagger/**
      swagger-ui:
        paths: classpath:META-INF/swagger/views/swagger-ui
        mapping: /swagger-ui/**
```

I expect the order of `swagger` and `swagger-ui` to not matter. Certainly, it should not cause failure to serve static resources.

### Actual Behaviour

The yaml file may not entirely control the order of mappings here: https://github.com/micronaut-projects/micronaut-servlet/blob/0b2f1d257e573cd7f1f8f29548aacc39c56123d9/http-server-jetty/src/main/java/io/micronaut/servlet/jetty/JettyFactory.java#L129-L136

In any case, if `path` is `/swagger-ui` and the first `mapping` is `/swagger`, then the path breaks from the `substring` call.

### Environment Information

Probably all versions of micronaut-servlet. Tested under 2.5.12 and visually inspected on 3.0.1-SNAPSHOT.

### Version

2.5.12, 3.0.1-SNAPSHOT

**Comment by mattmoss** [comment]
See #191 for additional details.

**Comment by timyates** [comment]
Fixed in 4.0.4 by https://github.com/micronaut-projects/micronaut-servlet/pull/555
